Tasks/JenkinsQueueJob/jobqueue.ts

```diff
@@ -0,0 +1,260 @@
+            if (job.state == JobState.Locating && (job.parent == null || job.parent.state == JobState.Done)) {
+                // group these together so only search is done per job name
+                if (!nextSearches[job.name]) {
+                    nextSearches[job.name] = this.searches[job.name];
+                }
+                nextSearches[job.name].searchFor(job);
+            }
+        }
+        return nextSearches;
+    }
+
+    getActiveJobs(): Job[] {
+        var active: Job[] = [];
+
+        for (var i in this.allJobs) {
+            var job = this.allJobs[i];
+            if (job.isActive()) {
+                active.push(job);
+            }
+        }
+
+        return active;
+    }
+
+    addJob(job: Job) {
+        if (this.allJobs.length == 0) {
+            this.rootJob = job;
+        }
+        this.allJobs.push(job);
+        if (this.searches[job.name] == null) {
+            this.searches[job.name] = new JobSearch(this, job.taskUrl, job.name);
+        }
+        job.search = this.searches[job.name];
+    }
+
+    flushJobConsolesSafely(): void {
+        if (this.findActiveConsoleJob() == null) { //nothing is currently writing to the console
+            var streamingJobs: Job[] = [];
+            var addedToConsole: boolean = false;
+            for (var i in this.allJobs) {
+                var job = this.allJobs[i];
+                if (job.state == JobState.Done) {
+                    if (!job.isConsoleEnabled()) {
+                        job.enableConsole(); // flush the finished ones
+                        addedToConsole = true;
+                    }
+                } else if (job.state == JobState.Streaming || job.state == JobState.Finishing) {
+                    streamingJobs.push(job); // these are the ones that could be running
+                }
+            }
+            // finally, if there is only one remaining, it is safe to enable its console
+            if (streamingJobs.length == 1) {
+                streamingJobs[0].enableConsole();
+            } else if (addedToConsole) {
+                for (var i in streamingJobs) {
+                    var job = streamingJobs[i];
+                    console.log('Jenkins job pending: ' + job.executableUrl);
+                }
+            }
+        }
+    }
+
+    /**
+     * If there is a job currently writing to the console, find it.
+     */
+    findActiveConsoleJob(): Job {
+        var activeJobs: Job[] = this.getActiveJobs();
+        for (var i in activeJobs) {
+            var job = activeJobs[i];
+            if (job.isConsoleEnabled()) {
+                return job;
+            }
+        }
+        return null;
+    }
+
+    findJob(name: string, executableNumber: number): Job {
+        for (var i in this.allJobs) {
+            var job = this.allJobs[i];
+            if (job.name == name && job.executableNumber == executableNumber) {
+                return job;
+            }
+        }
+        return null;
+    }
+
```

Comment:
I've tried tweeking the indent and the padding, but the net result is all or nothing, here's a snippet of what the mark down looks like:
<ul style="padding-left:0">
[TestChain_G #1268](http://xplatalm-linuxjenkins.southcentralus.cloudapp.azure.com/job/TestChain_G/1268) Success<br>
  <ul style="padding-left:2">
  [TestChain_H #1471](http://xplatalm-linuxjenkins.southcentralus.cloudapp.azure.com/job/TestChain_H/1471) Success<br>
    <ul style="padding-left:4">
    [TestChain_I #1690](http://xplatalm-linuxjenkins.southcentralus.cloudapp.azure.com/job/TestChain_I/1690) Success<br>
    </ul>
  </ul>
  <ul style="padding-left:2">
  [TestChain_I #1689](http://xplatalm-linuxjenkins.southcentralus.cloudapp.azure.com/job/TestChain_I/1689) Success<br>
  </ul>
</ul>